jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/Dispatcher.java

```diff
@@ -753,6 +754,36 @@ public Enumeration<String> getAttributeNames()
             names.add(AsyncContextState.ASYNC_QUERY_STRING);
             return Collections.enumeration(names);
         }
+
+        /**
+         * Get the parameters merged from the wrapped request, and the dispatch URI.
+         * If the dispatch URI is the same as the URI of the wrapped request, then
+         * we do not want to re-extract the query string, as the wrapped request will
+         * have already done the extraction.
+         * @return the request parameters
+         */
```

Comment:
Nice comment, but a couple of tweaks:
```suggestion
        /**
         * Get the parameters merged from the request wrapped by this {@code AsyncRequest}
         * and those in the {@code RequestDispatcher}.
         * If the target path is the same as the request path, then
         * we do not want to re-extract the query string, as the wrapped request will
         * have already done the extraction.
         * @return the request parameters
         */
```

Also a comment on the `AsyncRequest` class itself would be good